synapse/pipeline.yml

```diff
@@ -570,3 +570,46 @@ steps:
           # the hosts Docker.
           volumes:
             - "/var/run/docker.sock:/var/run/docker.sock"
+
+  - command:
+      # Build a docker image from the checked out Synapse source
+      - "docker build -t matrixdotorg/synapse:latest -f docker/Dockerfile ."
+      # Build a second docker image on top of the above image. This one sets up Synapse workers
+      # as well as everything else it needs to run a federating setup
+      - "docker build -t matrixdotorg/synapse:workers -f docker/Dockerfile-workers ."
+      # We use the complement:latest image to provide Complement's dependencies, but want
+      # to actually run against the master branch of Complement, so download it here.
+      - "wget https://github.com/matrix-org/complement/archive/master.tar.gz"
+      - "tar -xzf master.tar.gz"
+      # Build an image for running the above worker setup in Complement. This involves
+      # disabling rate-limiting, using Complement's CA etc.
+      - "docker build -t complement-synapse -f complement-master/dockerfiles/SynapseWorkers.Dockerfile complement-master/dockerfiles"
+      # Finally, compile and run the tests.
+      - "cd complement-master"
+      - "COMPLEMENT_BASE_IMAGE=complement-synapse:latest go test -v -tags synapse_blacklist ./tests"
+    label: "\U0001F9EA Complement | Synapse Workers"
+    agents:
+      # Running every worker takes a beefy system
+      queue: "xlarge"
+    plugins:
+      - docker#v3.7.0:
+          # The dockerfile for this image is at https://github.com/matrix-org/complement/blob/master/dockerfiles/ComplementCIBuildkite.Dockerfile.
+          image: "matrixdotorg/complement:latest"
+          mount-buildkite-agent: false
+          environment:
+            # Complement needs to know if it is running under CI
+            - "CI=true"
```

Comment:
:roll_eyes:

Reply:
https://github.com/matrix-org/complement/issues/60

I'm planning to change that myself once these PRs are in :)